django/contrib/admin/views/main.py

```diff
@@ -303,14 +303,57 @@ def get_ordering(self, request, queryset):
         # Add the given query's ordering fields, if any.
         ordering.extend(queryset.query.order_by)
 
-        # Ensure that the primary key is systematically present in the list of
+        # Ensure that a unique together set of fields is present in the list of
```

Comment:
I guess we could add a `get_total_ordering` method that gets passed `ordering` but I'm not sure if it shouldn't be the other way around: `get_queryset` calls `get_total_ordering` that calls `get_ordering`. It seems like it would make more sense than the `get_queryset(get_ordering(get_total_ordering))` chain.

Would you like me to create a new ticket for this feature and continue work in this PR?

Reply:
Errr... 😃 it’s part of the old ticket in my view. (That may not be canonical.)

A new ticket might help in case we need multiple reviews, but having come across this today, I kind of thought it would be good to get in for 2.2 if that’s realistic on you end. 

Tl;dr: don’t really mind. 

My initial thought on the code structure would be to go for the minimal change here. Add the helper (even private) to encapsulate this logic, and call it quits. We can always allow a restructuring later. 

However, if you have ideas for something better I wouldn’t object. 🙂

Reply:
OK, I see. 

I might be tempted to do it the other way round. Main reason is that if I override `get_total_ordering()` I need to call `get_ordering()` as the first step, whereas if `get_total_ordering()` just takes and returns the `ordering` it's freestanding. 

Not strongly inclined though. I'm happy to go with your preferred option. 

Do you want to document it, maybe as a `Fixed #29943 -- ...`?